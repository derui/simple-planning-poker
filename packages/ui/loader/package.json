{
  "name": "@spp/ui-loader",
  "version": "1.0.0",
  "description": "",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "scripts": {
    "check-type": "tsc --noEmit",
    "build:tsc": "tsc",
    "dev:build:tsc": "tsc --watch --noEmit",
    "build:css": "tailwindcss -i src/style.css -o dist/style.css",
    "dev:build:css": "tailwindcss --watch -i src/style.css -o dist/style.css",
    "test": "vitest run",
    "dev:test": "vitest watch",
    "lint": "eslint --cache --cache-location=node_modules/.cache/eslint --fix src/",
    "dev:storybook": "storybook dev -p 6006",
    "build:storybook": "storybook build",
    "dev": "concurrently \"pnpm:dev:*\"",
    "build": "concurrently \"pnpm:build:*\""
  },
  "type": "module",
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@spp/eslint-config": "workspace:^",
    "@spp/tailwind-config": "workspace:^",
    "@spp/typescript-config": "workspace:^",
    "@spp/ui-icon": "workspace:^",
    "@storybook/addon-controls": "catalog:",
    "@storybook/cli": "catalog:",
    "@storybook/react": "catalog:",
    "@storybook/react-vite": "catalog:",
    "@testing-library/react": "catalog:",
    "@types/react": "catalog:",
    "@vitejs/plugin-react": "catalog:",
    "concurrently": "catalog:",
    "eslint": "catalog:",
    "jsdom": "catalog:",
    "react-dom": "catalog:",
    "tailwindcss": "catalog:",
    "typescript": "catalog:",
    "vite": "catalog:",
    "vitest": "catalog:"
  },
  "dependencies": {
    "@spp/shared-color-variant": "workspace:^",
    "clsx": "catalog:",
    "react": "catalog:"
  }
}
