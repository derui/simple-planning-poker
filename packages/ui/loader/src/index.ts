export { Loader } from "./loader.js";
