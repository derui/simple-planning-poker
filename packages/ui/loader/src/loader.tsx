import { VariantName } from "@spp/shared-color-variant";
import { Icon } from "@spp/ui-icon";
import { hidden, loader, spinLoader } from "./style.css.js";

type Size = "s" | "m" | "l";

interface Props {
  shown: boolean;
  size: Size;
  variant?: VariantName;
}

// eslint-disable-next-line func-style
export function Loader({ size, shown, variant }: Props) {
  const root = shown ? loader[size] : hidden;
  const spinner = spinLoader[size];

  return (
    <span className={root} data-shown={shown} role="alert">
      <span className={spinner}>
        <Icon.Loader2 size={size} variant={variant} />
      </span>
    </span>
  );
}
