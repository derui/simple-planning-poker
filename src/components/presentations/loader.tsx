import classNames from "classnames";
import { BaseProps, generateTestId } from "../base";
import { iconize } from "../iconize";

type Size = "s" | "m" | "l";

interface Props extends BaseProps {
  shown: boolean;
  size: Size;
}

const sizeStyle = (size: Size) =>
  classNames(
    {
      "w-5": size === "s",
      "h-5": size === "s",
    },
    {
      "w-6": size === "m",
      "h-6": size === "m",
    },
    {
      "w-7": size === "l",
      "h-7": size === "l",
    }
  );

const styles = {
  root: (size: Size, shown: boolean) =>
    classNames(
      "relative",
      {
        "inline-block": shown,
        hidden: !shown,
      },
      sizeStyle(size)
    ),
  spinLoader: (size: Size) =>
    classNames(
      "absolute",
      "top-0",
      "left-0",
      "inline-block",
      "z-10",
      iconize("loader-2", { size }),
      "before:bg-gray",
      "animate-spin",
      sizeStyle(size)
    ),
};

// eslint-disable-next-line func-style
export function Loader({ size, shown, testid }: Props) {
  const gen = generateTestId(testid);

  return (
    <span className={styles.root(size, shown)} data-testid={gen("root")} data-shown={shown}>
      <span className={styles.spinLoader(size)}></span>
    </span>
  );
}
