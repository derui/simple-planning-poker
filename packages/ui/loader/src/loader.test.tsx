import { test, afterEach, expect } from "vitest";
import { render, cleanup, screen } from "@testing-library/react";
import { Loader } from "./loader.js";

afterEach(cleanup);

test("should be able to render", async () => {
  render(<Loader size="s" shown={false} />);

  const ret = screen.queryByRole("alert");

  expect(ret).not.toBeNull();
  expect(ret?.dataset).toHaveProperty("shown", "false");
});

test("display loader", async () => {
  render(<Loader size="s" shown={true} />);

  const ret = screen.queryByRole("alert");

  expect(ret).not.toBeNull();
  expect(ret?.dataset).toHaveProperty("shown", "true");
});
