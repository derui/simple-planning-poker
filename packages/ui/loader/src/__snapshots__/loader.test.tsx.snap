// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`display loader with variant 1`] = `
<div>
  <span
    class="style_loader_s__1v4i49g2 style_base__1v4i49g0"
    data-shown="true"
    role="alert"
  >
    <span
      class="style_spinLoader_s__1v4i49g6 style_spinLoaderBase__1v4i49g5 style_animation_spin__6vbqbm3r"
    >
      <span
        class="loader2_small_cerise__1x64t376 style_iconSmall__12cnpf31 style_iconBase__12cnpf30"
      />
    </span>
  </span>
</div>
`;
